parser-typechecker/src/U/Codebase/Branch/Diff.hs

```diff
@@ -172,3 +189,27 @@ nameChanges namePrefix (TreeDiff (DefinitionDiffs {termDiffs, typeDiffs} :< chil
       xs
         & Set.toList
         & fmap (name,)
+
+-- | Get a 'NameBasedDiff' from a 'TreeDiff'.
+nameBasedDiff :: TreeDiff -> NameBasedDiff
+nameBasedDiff (TreeDiff (DefinitionDiffs {termDiffs, typeDiffs} :< children)) =
```

Comment:
TreeDiff just wraps a Cofree, so I believe you can factor out the recursion into a single foldMap here if you like.

```suggestion
nameBasedDiff (TreeDiff d) =
  d & foldMap \defDiff -> 
        NameBasedDiff 
        { terms = nameBasedTermDiff defDiff
        , types = nameBasedTypeDiff defDiff
        }
```
                
```